aten/src/ATen/native/mkldnn/Matmul.cpp

```diff
@@ -402,6 +409,70 @@ bool use_mkldnn_matmul(
   return (use_mkldnn_bf16_matmul(mat1, mat2, result) || use_mkldnn_fp16_matmul(mat1, mat2, result) || use_mkldnn_bf32_matmul(mat1, mat2, result));
 }
 
+void mkldnn_matmul_i8i8i32(
+    const Tensor &mat1,
+    const Tensor &mat2,
+    const Tensor &result) {
+  // x:s8 * w:s8 -> y:s32
+  TORCH_CHECK(
+      mat1.scalar_type() == c10::kChar,
+      __func__, ": mat1 dtype should be signed int8 but found ",
+      mat1.scalar_type());
+  TORCH_CHECK(
+      mat2.scalar_type() == c10::kChar,
+      __func__, ": mat2 dtype should be signed int8 but found ",
+      mat2.scalar_type());
+  TORCH_CHECK(
+      mat1.dim() == 2 && mat2.dim() == 2,
+      __func__, ": Expect mat1 and mat2 are 2d but got ",
+      mat1.dim(),
+      " and ",
+      mat2.dim());
+  TORCH_CHECK(
+      mat1.size(1) == mat2.size(0),
+      "matmul_i8i8i32: mat1 shape and mat2 shape do not match, got ",
+      mat1.sizes(),
+      " and ",
+      mat2.sizes());
+  // Create ideep tensors for oneDNN computation
+  auto src = ideep::tensor(
+      {mat1.sizes().vec(),
+       ideep::tensor::data_type::s8,
+       mat1.strides().vec()},
+      mat1.data_ptr());
+  auto wei = ideep::tensor(
+      {mat2.sizes().vec(),
+       ideep::tensor::data_type::s8,
+       mat2.strides().vec()},
+      mat2.data_ptr());
+  auto dst = ideep::tensor(
+      {result.sizes().vec(),
+       ideep::tensor::data_type::s32,
+       result.strides().vec()},
+      result.data_ptr());
+  // Create primitive desc
```

Comment:
Thanks for the suggestion. I have run benchmarks locally to compare the implementations with BLAS API and primitive API. In most cases, BLAS API showed better performance. However, the BLAS API requires input buffers to be contiguous. So, the current dispatching rule is that if input buffers are contiguous, the BLAS API is used; otherwise, the primitive API is used. Do you think it's OK? Thanks.